.github/workflows/publish-cosign-sample.yml

```diff
@@ -0,0 +1,61 @@
+name: publish-cosign-sample
+
+on: 
+ workflow_dispatch:
+   
+env:
+  REGISTRY: ghcr.io
+  
+permissions:
+  contents: read
+
+jobs:
+  build-publish:
+    name: "Build and publish cosign signed sample image"
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    permissions:
+      contents: write
+      packages: write
+      id-token: write
+    steps:            
+      - name: Harden Runner
+        uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
+        with:
+          egress-policy: audit
+
+      - name: Install cosign
+        uses: sigstore/cosign-installer@59acb6260d9c0ba8f4a2f9d9b48431a222b68e20 # v3.5.0
+
+      - name: Get repo
+        run: |          
+          echo "REPOSITORY=${{ env.REGISTRY }}/${{ github.repository }}" >> $GITHUB_ENV
+
+      - name: Write signing key to disk
+        run: 'echo "$KEY" > cosign.key'
+        shell: bash
+        env:
+          KEY: ${{ secrets.COSIGN_PRIVATE_KEY }}
+
+      - name: Log in to GHCR
+        uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build sampled signed images
+        run: |
+          docker build -t ${REPOSITORY}/cosign-image:signed-key https://github.com/wabbit-networks/net-monitor.git
+          docker build --no-cache -t ${REPOSITORY}/cosign-image:signed-keyless https://github.com/wabbit-networks/net-monitor.git
+      - name: Push signed images
+        run: |
+          docker push ${REPOSITORY}/cosign-image:signed-key
+          docker push ${REPOSITORY}/cosign-image:signed-keyless
+      - name: Sign image with cosign
+        run: |
+          cosign sign --yes --key cosign.key ${REPOSITORY}/cosign-image:signed-key
+          cosign sign --yes ${REPOSITORY}/cosign-image:signed-keyless
+      - name: Push unsigned image
+        run: 
+          oras cp mcr.microsoft.com/mcr/hello-world:latest ${REPOSITORY}/cosign-image:unsigned
```

Comment:
just curious if hello-world was built from net-monitor

Reply:
It's not currently. I can switch it to net-monitor. The goal was just to find an image that will result in a unique digest so that signatures don't get attached to the same digest. I can build with `--no-cache` and it should generate a unique image.